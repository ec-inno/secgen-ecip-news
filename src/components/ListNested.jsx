import React from 'react';
import ListItemNested from './ListItemNested';

const ListNested = ({ list }) =>
  list && Object.keys(list).length ? (
    <ul className="ecl-unordered-list">
      {Object.keys(list).map((item, id) => (
        <ListItemNested item={list[item]} id={id} />
      ))}
    </ul>
  ) : (
    ''
  );

export default ListNested;
