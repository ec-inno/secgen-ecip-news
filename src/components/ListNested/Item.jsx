import React from 'react';
import PropTypes from 'prop-types';
import { Link as LinkInternal } from 'gatsby';
import LinkExternal from '../Link/LinkEcl';

const Item = ({ item, id }) => {
  const accepted = ['href', 'title', 'external'];

  if (!item) return '';

  let ul = null;
  const children = {};

  Object.keys(item).forEach(prop => {
    if (!accepted.includes(prop)) {
      children[prop] = item[prop];
    }
  });

  if (Object.keys(children).length) {
    ul = (
      <ul className="ecl-unordered-list">
        {Object.keys(children).map((childPath, key) => (
          <Item item={children[childPath]} key={key} />
        ))}
      </ul>
    );
  }

  return (
    <li className="ecl-unordered-list__item" key={id}>
      {item.external ? (
        <LinkExternal
          href={item.href}
          className="ecl-link ecl-link--standalone"
        >
          {item.title}
        </LinkExternal>
      ) : (
        <LinkInternal to={item.href} className="ecl-link ecl-link--standalone">
          {item.title}
        </LinkInternal>
      )}
      {ul}
    </li>
  );
};

Item.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string,
    href: PropTypes.string,
    external: PropTypes.bool,
  }),
  id: PropTypes.string,
};

export default Item;
