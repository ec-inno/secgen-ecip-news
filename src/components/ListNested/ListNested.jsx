import React from 'react';
import Item from './Item';

const ListNested = ({ list }) => {
  if (!list || Object.keys(list).length === 0) {
    return '';
  }

  return (
    <ul className="ecl-unordered-list">
      {Object.keys(list)
        .sort()
        .map((item, id) => (
          <Item item={list[item]} key={id} id={id} />
        ))}
    </ul>
  );
};

export default ListNested;
