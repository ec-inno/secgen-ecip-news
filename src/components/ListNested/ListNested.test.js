import React from 'react';
import renderer from 'react-test-renderer';
import ListNested from './ListNested';

describe('ListNested', () => {
  it('Returns an empty string on missing input', () => {
    const tree = renderer.create(<ListNested />).toJSON();
    expect(tree).toBe('');
  });
});
