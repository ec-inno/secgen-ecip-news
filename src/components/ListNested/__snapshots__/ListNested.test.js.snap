// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListNested Returns an empty string on missing input 1`] = `""`;
